sql/core/src/test/scala/org/apache/spark/sql/connector/DeleteFromTableSuiteBase.scala

```diff
@@ -63,6 +63,31 @@ abstract class DeleteFromTableSuiteBase extends RowLevelOperationSuiteBase {
         Row(6, "hr", "new-text")))
   }
 
+  test("delete from table with table constraints") {
+    sql(
+      s"""
+         |CREATE TABLE $tableNameAsString (
+         | pk INT NOT NULL PRIMARY KEY,
+         | id INT UNIQUE,
+         | dep STRING,
+         | CONSTRAINT pk_check CHECK (pk > 0))
```

Comment:
DELETE does not add new table, so ideally constraints doesn't apply here?

Reply:
yes, ideally we should skip adding the check constraint filter for DELETE.
This PR is more of adding tests to ensure delete works with constraints.